File: OhanaServer/src/main/java/com/CentroOhana/OhanaServer/Services/AuthService.java
package com.CentroOhana.OhanaServer.Services;

import com.CentroOhana.OhanaServer.Config.JwtService;
import com.CentroOhana.OhanaServer.Controllers.AuthRequest;
import com.CentroOhana.OhanaServer.Controllers.AuthResponse;
import com.CentroOhana.OhanaServer.Models.User;
import com.CentroOhana.OhanaServer.Repositories.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final UserRepository userRepository;
    private final JwtService jwtService;
    private final PasswordEncoder passwordEncoder;
    private final AuthenticationManager authenticationManager;


    public AuthResponse login(AuthRequest request) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword()));
        UserDetails user=userRepository.findByUsername(request.getUsername()).orElseThrow();
        String token=jwtService.getToken(user);
        return AuthResponse.builder()
                .token(token)
                .build();

    }

    public AuthResponse register(AuthRequest request) {
        User user = User.builder()
                .username(request.getUsername())
                .password(request.getPassword())
                .build();

        userRepository.save(user);

        return AuthResponse
                .builder()
                .token(jwtService.getToken(user))
                .build();
    }
}

File: front/web-ohana/src/services/authService.ts
import { AuthRequest } from "../models/auth-request";
import { AuthResponse } from "../models/auth-response";
import ApiService from "./apiService";

const AuthService = {
  login: async (request: AuthRequest): Promise<AuthResponse> => {
    const response = await ApiService.post<AuthResponse>("auth/login", {
      username: request.username,
      password: request.password,
    });

    if (!response.success || !response.data?.token) {
      throw new Error("Login failed: Token not received");
    }

    localStorage.setItem("token", response.data.token);
    return response.data;
  },

  register: async (request: AuthRequest): Promise<AuthResponse> => {
    const response = await ApiService.post<AuthResponse>("auth/register", {
      username: request.username,
      password: request.password,
    });

    console.log("Register response:", response);
    localStorage.setItem("token", response.data.token);
    return response.data;
  },
};

export default AuthService;
